### PR TITLE
Grap 23 initialize firebase setup in project

### PR DESCRIPTION
**Pull request by 1LisaW**
add firebase packegas to project.
Create Provider and custom Hook for auth data. Add auth functionality (firebase) to auth forms.

### CODE REVIEW

File: src/components/FormInput/FormInput.test.tsx
Comment (ppmarkek) [inline src/components/FormInput/FormInput.test.tsx:1]: Create new folder for test and create for reused components create folder atoms
image

File: src/pages/SignIn/SignIn.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      email: '',
+      password: '',
+      passwordConfirm: '',
+    },
+    onSubmit: (values) => {
+      alert(JSON.stringify(values, null, 2));
+    },
+  });
+  return (
+    <Grid
```
Comment (ppmarkek) [inline src/pages/SignIn/SignIn.tsx:25]: Use 'styled mui' for component styling

File: src/pages/SignIn/SignIn.tsx
Comment (ppmarkek) [inline src/pages/SignIn/SignIn.tsx:1]: Create new folder for test

File: src/pages/SignUp/SignUp.test.tsx
Comment (ppmarkek) [inline src/pages/SignUp/SignUp.test.tsx:1]: Create new folder for test

File: src/pages/SignUp/SignUp.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      password: '',
+      passwordConfirm: '',
+    },
+    onSubmit: (values) => {
+      registerWithEmailAndPassword(values.name, values.email, values.password);
+      alert(JSON.stringify(values, null, 2));
+    },
+  });
+  return (
+    <Grid
```
Comment (ppmarkek) [inline src/pages/SignUp/SignUp.tsx:31]: Create new folder for test

**Review by svyatoslavkk — APPROVED** [review]

**Review by ppmarkek — APPROVED** [review]

**Pull request merged**